chips/nrf52/src/ble_connection/ble_advertising_hil.rs

```diff
@@ -0,0 +1,278 @@
+    DataChannel18 = 42,
+    DataChannel19 = 44,
+    DataChannel20 = 46,
+    DataChannel21 = 48,
+    DataChannel22 = 50,
+    DataChannel23 = 52,
+    DataChannel24 = 54,
+    DataChannel25 = 56,
+    DataChannel26 = 58,
+    DataChannel27 = 60,
+    DataChannel28 = 62,
+    DataChannel29 = 64,
+    DataChannel30 = 66,
+    DataChannel31 = 68,
+    DataChannel32 = 70,
+    DataChannel33 = 72,
+    DataChannel34 = 74,
+    DataChannel35 = 76,
+    DataChannel36 = 78,
+    AdvertisingChannel37 = 2,
+    AdvertisingChannel38 = 26,
+    AdvertisingChannel39 = 80,
+}
+
+impl RadioChannel {
+    pub fn get_next_advertising_channel(&self) -> Option<RadioChannel> {
+        match *self {
+            RadioChannel::AdvertisingChannel37 => Some(RadioChannel::AdvertisingChannel38),
+            RadioChannel::AdvertisingChannel38 => Some(RadioChannel::AdvertisingChannel39),
+            _ => None,
+        }
+    }
+    pub fn get_channel_index(&self) -> u32 {
+        match *self {
+            RadioChannel::DataChannel0 => 0,
+            RadioChannel::DataChannel1 => 1,
+            RadioChannel::DataChannel2 => 2,
+            RadioChannel::DataChannel3 => 3,
+            RadioChannel::DataChannel4 => 4,
+            RadioChannel::DataChannel5 => 5,
+            RadioChannel::DataChannel6 => 6,
+            RadioChannel::DataChannel7 => 7,
+            RadioChannel::DataChannel8 => 8,
+            RadioChannel::DataChannel9 => 9,
+            RadioChannel::DataChannel10 => 10,
+            RadioChannel::DataChannel11 => 11,
+            RadioChannel::DataChannel12 => 12,
+            RadioChannel::DataChannel13 => 13,
+            RadioChannel::DataChannel14 => 14,
+            RadioChannel::DataChannel15 => 15,
+            RadioChannel::DataChannel16 => 16,
+            RadioChannel::DataChannel17 => 17,
+            RadioChannel::DataChannel18 => 18,
+            RadioChannel::DataChannel19 => 19,
+            RadioChannel::DataChannel20 => 20,
+            RadioChannel::DataChannel21 => 21,
+            RadioChannel::DataChannel22 => 22,
+            RadioChannel::DataChannel23 => 23,
+            RadioChannel::DataChannel24 => 24,
+            RadioChannel::DataChannel25 => 25,
+            RadioChannel::DataChannel26 => 26,
+            RadioChannel::DataChannel27 => 27,
+            RadioChannel::DataChannel28 => 28,
+            RadioChannel::DataChannel29 => 29,
+            RadioChannel::DataChannel30 => 30,
+            RadioChannel::DataChannel31 => 31,
+            RadioChannel::DataChannel32 => 32,
+            RadioChannel::DataChannel33 => 33,
+            RadioChannel::DataChannel34 => 34,
+            RadioChannel::DataChannel35 => 35,
+            RadioChannel::DataChannel36 => 36,
+            RadioChannel::AdvertisingChannel37 => 37,
+            RadioChannel::AdvertisingChannel38 => 38,
+            RadioChannel::AdvertisingChannel39 => 39,
+        }
+    }
+
+    pub fn from_channel_index(index: u8) -> Option<RadioChannel> {
```

Comment:
would be nice to use the `TryFrom` trait here instead, then we could use `x.into()`